File: codeforces/ungrouped/cf1999d.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

bool solve(string s, string t, string &res) {
    int s_ptr = s.size() - 1;
    int t_ptr = t.size() - 1;
    res = "";

    while (t_ptr >= 0 && s_ptr >= 0) {
        if (s[s_ptr] == t[t_ptr] || s[s_ptr] == '?') {
            res += t[t_ptr];
            t_ptr--;
        } else {
            res += s[s_ptr];
        }
        s_ptr--;
    }

    // 如果t还有未匹配的字符，说明无法匹配
    if (t_ptr >= 0) {
        return false;
    }

    // 剩余的s如果有"?"，替换为"a"
    while (s_ptr >= 0) {
        res += (s[s_ptr] == '?' ? 'a' : s[s_ptr]);
        s_ptr--;
    }

    // 反转结果字符串
    reverse(res.begin(), res.end());
    return true;
}

int main() {
    int num_of_tc;
    cin >> num_of_tc;

    for (int i = 0; i < num_of_tc; ++i) {
        string s, t;
        cin >> s >> t;
        string res;
        if (solve(s, t, res)) {
            cout << "Yes" << endl;
            cout << res << endl;
        } else {
            cout << "No" << endl;
        }
    }

    return 0;
}

File: codeforces/ungrouped/cf1999d.py
# def solve(s, t):
#     res = ""
#     idx = 0
#     for i in t:
#         while idx < len(s) and s[idx] != i and s[idx] != "?":
#             res += s[idx]
#             idx += 1
#         if idx >= len(s):
#             yield "No"
#             return
#         if s[idx] == "?":
#             res += i
#             idx += 1
#     while idx < len(s):
#         res += s[idx] if s[idx] != "?" else "a"
#         idx += 1

#     yield "Yes"
#     # for i in range(len(s)):
#     #     if s[i] == "?":
#     #         s[i] = "a"

#     yield res


def solve(s, t):
    s_ptr = len(s) - 1
    t_ptr = len(t) - 1
    res = ""
    while t_ptr >= 0 and s_ptr >= 0:
        if s[s_ptr] == t[t_ptr] or s[s_ptr] == "?":
            res += t[t_ptr]
            t_ptr -= 1
        else:
            res += s[s_ptr]
        s_ptr -= 1
    if t_ptr >= 0:
        yield "No"
        return
    while s_ptr >= 0:
        res += s[s_ptr] if s[s_ptr] != "?" else "a"
        s_ptr -= 1
    yield "Yes"
    yield res[::-1]


num_of_tc = int(input())
for _ in range(num_of_tc):
    s = input()
    t = input()
    for i in solve(s, t):
        print(i)
